### DISCUSSION TITLE
[React-Native] Keyboard disappear after entering text in text input

### DISCUSSION THREAD

**Discussion by nguyenduckhai** [react-hook-form/react-hook-form#5864] [Q&A] [answered]
**Describe the bug**
The keyboard disappears after entering text in text input.  To be precise, I have 2 text inputs, both are required. After I entered the 1st one, ignore the 2nd one, and press the submit button, the 2nd field shows the required message (which is expected). When I am typing the 2nd text input, the keyboard immediately disappears after the 1st character.

**To Reproduce**

1. Go to my expo snack
2. Enter text in the first text input
3. Press the button which it shows the required message
4. Enter text in the second text input.
5. Keyboard disappears while entering text.

**Codesandbox link (Required)**

* [My expo snack](https://snack.expo.io/@closebeta12/keyboard-bug) 

**Expected behavior**
The keyboard does not disappear while entering text in text input

**Smartphone (please complete the following information):**

- Device: Redmi note 8 pro
- OS: Android 10

**Environment**

| Tech            | Version |
|-----------------|---------|
| React-hook-form | 7.10.1  |
| Expo            | 40.0.0  |

**Comment by bluebill1049** [comment, marked as answer]
This is not related to this library, as react hook form does nothing toward the virtual keyboard.

**Reply by nguyenduckhai** [reply]
I understand. 
It's kinda weird that this bug only happens if I'm using a common controller component like this
```js
const Input = ({ name, defaultValue }) => (
        <Controller
            control={control}
            rules={{
                required: true,
            }}
            render={({ field: { onChange, onBlur, value } }) => (
                <TextInput
                    style={styles.input}
                    onBlur={onBlur}
                    onChangeText={onChange}
                    value={value}
                />
            )}
            name={name}
            defaultValue={defaultValue}
        />
  );
  
  return (
    <View style={styles.container}>
      <View style={{paddingBottom: 20}}>
        <Input name="firstName" defaultValue=""/>
        {errors.firstName && <Text style={styles.label}>is required</Text>}
      </View>

      <View style={{paddingBottom: 20}}>
        <Input name="lastName" defaultValue=""/>
        {errors.lastName && <Text style={styles.label}>is required</Text>}
      </View>

      <View style={styles.button}>
        <Button
          style={styles.buttonInner}
          color
          title="Button"
          onPress={handleSubmit(onSubmit)}
        />
      </View>
    </View>
  );
};
```
If I'm rewriting them like this, this will work without the bug
```js
return (
    <View style={styles.container}>
      <View style={{paddingBottom: 20}}>
        <Controller
            control={control}
            rules={{
                required: true,
            }}
            render={({ field: { onChange, onBlur, value } }) => (
                <TextInput
                    style={styles.input}
                    onBlur={onBlur}
                    onChangeText={onChange}
                    value={value}
                />
            )}
            name='firstName'
            defaultValue=''"
        />
        {errors.firstName && <Text style={styles.label}>is required</Text>}
      </View>

      <View style={{paddingBottom: 20}}>
        <Controller
            control={control}
            rules={{
                required: true,
            }}
            render={({ field: { onChange, onBlur, value } }) => (
                <TextInput
                    style={styles.input}
                    onBlur={onBlur}
                    onChangeText={onChange}
                    value={value}
                />
            )}
            name='lastName'
            defaultValue=''"
        />
        {errors.lastName && <Text style={styles.label}>is required</Text>}
      </View>
   </View>
);
```
I really don't know how to deal with this bug since I work with 8 inputs in my project and I don't want to rewrite all of them like the above. I hope you have other solution for me to deal with them.

**Reply by bluebill1049** [reply]
move that component out, i think you are mount and unmont your input.

**Reply by nguyenduckhai** [reply]
Thank you! It's work! 🙏
To clarify what I understand, what you're saying means that your custom hook "useForm" contains some default hook from react native to mount and unmount components inside. 
In my case, this will mount and unmount my Input component if the errors verify whether there is a text or not. 
To solve this, I have to move my component outside the main component which will not effected by the useForm. Is my understanding correct ?
